src/components/QuranReader/index.tsx

```diff
@@ -1,23 +1,39 @@
 import React from 'react';
 import { useSelector } from 'react-redux';
+import ChapterType from 'types/ChapterType';
+import { fetcher } from 'src/api';
+import { useSWRInfinite } from 'swr';
+import { makeUrl } from 'src/utils/api';
 import VerseType from '../../../types/VerseType';
 import QuranPageView from './QuranPageView';
 import TranslationView from './TranslationView';
 import { selectReadingView } from '../../redux/slices/QuranReader/readingView';
 import { ReadingView } from './types';
 
 type QuranReaderProps = {
-  verses: VerseType[];
+  initialData: { verses: VerseType[] };
+  chapter: ChapterType;
 };
 
-const QuranReader = ({ verses }: QuranReaderProps) => {
+const QuranReader = ({ initialData, chapter }: QuranReaderProps) => {
+  const { data } = useSWRInfinite(
+    (index) => {
+      return makeUrl(`/chapters/${chapter.id}/verses`, { translations: 20, page: index }); // TODO: select the translation using the user preference
```

Comment:
In the follow-up PR, `index` will represent that API page we're fetching using the `size` and `setSize` methods from useSWRInfinite